scripts/auth.js

```diff
@@ -0,0 +1,184 @@
+document.addEventListener("DOMContentLoaded", function () {
+  const form = document.querySelector("form.auth-form");
+  if (!form) return;
+  const button = form.querySelector("button.auth-button");
+  button.disabled = true;
+
+  // 전체 폼 유효성 검사
+  function checkFormValidity() {
+    let valid = true;
+    const inputs = form.querySelectorAll("input");
+    inputs.forEach(function (input) {
+      if (input.value.trim() === "") {
+        valid = false;
+      }
+    });
+    if (form.querySelectorAll(".error-message").length > 0) {
+      valid = false;
+    }
+    button.disabled = !valid;
+  }
```

Comment:
💊 제안
폼의 유효성을 검사하는 로직을 따로 분리해주신 것은 좋습니다. 
다만 이미 validateEmail과 같은 각 인풋별 유효성 검사 함수를 선언해두셨으니 이를 이용해서 유효성 검사를 하시거나
유효하지 않다면 error-message 라는 클래스를 가진 태그가 폼안에 존재할테니 그냥 해당 여부만 확인하시면 될 것 같습니다!

```suggestion
  function checkFormValidity() {
    const isValid = form.querySelectorAll(".error-message").length > 0;

    button.disabled = !isValid;
  }
```